setup/includes/upgrades/common/3.0.0-update-upload_files-upload_images.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Common upgrade script for modify upload_files, upload_images System Setting
+ *
+ * @var modX $modx
+ * @package setup
+ */
+
+$messageTemplate = '<p class="%s">%s</p>';
+
+$keys = ['upload_files','upload_images'];
+
+foreach ($keys as $key) {
+    $success = false;
+    
+    /** @var modSystemSetting $setting */
+    $setting = $modx->getObject('modSystemSetting', array('key' => $key));
+    if ($setting) {
+        $value = $setting->get('value');
+        $tmp = explode(',', $value);
+        if (in_array('webp', $tmp)) {
+            continue;
+        } else {
+            $setting->set('value', $value . ',webp');
+            if ($setting->save()) {
+                $success = true;
+            }
+        }
+    }
+    
+    if ($success) {
+        $this->runner->addResult(modInstallRunner::RESULT_SUCCESS,
+            sprintf($messageTemplate, 'ok', $this->install->lexicon('system_setting_update_success', ['key' => $key])));
+    } else {
```

Comment:
If I'm reading the code correctly, this `else` will also cause a warning to be shown when the settings already contained the webp value - that doesn't seem ideal?

Reply:
No. `foreach` will continue if the settings already contained the webp value and message will not shown:
https://github.com/modxcms/revolution/blob/17b38843490fe6608e5d7475e83915c72f3acf03/setup/includes/upgrades/common/3.0.0-update-upload_files-upload_images.php#L21-L23

Reply:
$success stays false then. Mark is right.

Reply:
At the moment `$success` gets true, when one setting is saved. I suggest to set it to false if one save is not successful (and start with a true value).

Reply:
The `if ($success)` is part of the foreach, if `webp` is already part of the setting it will never reach the `if ($success)`.

Reply:
OMG. A bit unexpected code for me.